addons/binding/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/handler/BoschIndegoHandler.java

```diff
@@ -0,0 +1,251 @@
+            if (command instanceof DecimalType) {
+                synchronized (this) {
+                    commandToSend = ((DecimalType) command).intValue();
+                    this.notifyAll();
+                }
+            } else {
+            }
+
+            // Note: if communication with thing fails for some reason,
+            // indicate that by setting the status with detail information
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+            // "Could not control device at IP address x.x.x.x");
+        }
+    }
+
+    @Override
+    public void updateConfiguration(Configuration configuration) {
+        // TODO Auto-generated method stub
+        super.updateConfiguration(configuration);
+
+    }
+
+    private synchronized void poll() throws InterruptedException {
+        // Create controller instance
+        try {
+            IndegoController controller = new IndegoController(getConfig().get("username").toString(),
+                    getConfig().get("password").toString());
+            // Connect to server
+            controller.connect();
+            // Query the device state
+            DeviceStateInformation state = controller.getState();
+            DeviceStatus statusWithMessage = DeviceStatus.decodeStatusCode(state.getState());
+            int eshStatus = getEshStatusFromCommand(statusWithMessage.getAssociatedCommand());
+            int mowed = state.getMowed();
+            int error = state.getError();
+            int statecode = state.getState();
+            boolean ready = isReadyToMow(state.getState(), state.getError());
+
+            if (verifyCommand(commandToSend, eshStatus, state.getState(), error)) {
+                logger.debug("Sending command...");
+                updateState(TEXTUAL_STATE, new StringType("Refreshing..."));
+                controller.sendCommand(getCommandFromEshStatus(commandToSend));
+                commandToSend = 0;
+                try {
+                    for (int i = 0; i < 30 && !Thread.interrupted(); i++) {
+                        DeviceStateInformation stateTmp = controller.getState();
+                        if (state.getState() != stateTmp.getState()) {
+                            state = stateTmp;
+                            statusWithMessage = DeviceStatus.decodeStatusCode(state.getState());
+                            eshStatus = getEshStatusFromCommand(statusWithMessage.getAssociatedCommand());
+                            mowed = state.getMowed();
+                            error = state.getError();
+                            statecode = state.getState();
+                            ready = isReadyToMow(state.getState(), state.getError());
+                            break;
+                        }
+                        Thread.sleep(1000);
+                    }
+
+                } catch (InterruptedException e) {
+                    controller.disconnect();
+                    throw e;
```

Comment:
Are you sure you want to rethrow the InterruptedException without resetting the interrupted flag in the thread using `Thread.currentThread().interrupt();`?  (The answer depends on the caller's expectation.  [Here](http://www.yegor256.com/2015/10/20/interrupted-exception.html) is a summary of the subject.)